docs/diagrams/BasicRecipeDiagram.puml

```diff
@@ -0,0 +1,125 @@
+@startuml BasicRecipeDiagram
+show members
+
+hide circle
+skinparam classAttributeIconSize 0
+
+class RecipeBook {
+    -recipes : UniqueRecipeList
+    +list()
+    +duplicate()
+}
+
+RecipeBook "1..*" *--> "many" Recipe : contains >
+
+class Recipe {
+    -name : String
+    -ingredients : IngredientList
+    -instructions : InstructionList
+    -tags : TagList
+    +modify()
+}
+
+class IngredientList {
+    -data : ArrayList<Ingredient>
+    +add()
+    +set()
+    +remove()
+    +list()
+}
+
+class Ingredient {
+    -name : String
+    -quantity : Quantity
+}
+
+IngredientStore <|--> Freezer : is a very cold & dry <
+IngredientStore <|--> Refrigerator : is a cold <
+IngredientStore <|--> Larder : is a cool, dark & dry <
+IngredientStore "0..*" o-r-> "many" Ingredient : stores <
+
+class InstructionList {
+    -data : HashMap<Instruction>
+    +add()
+    +edit()
+    +list()
+    +remove()
+}
+
+class Instruction {
+    -detail : String
+}
+
+class TagList {
+    -data : ArrayList<Tag>
+    +add()
+    +remove()
+}
+
+class Tag {
+    -name : String
+}
+
+TagList "0..*" o--> "many" Tag : contains >
+
+
+class Quantity {
+    -amount : Float
+    -unit : Unit
+}
+
+class Unit {
+    -name : String
+    -symbol : UnitSymbol
+    {static} +convertTo()
+}
+
+enum UnitSymbol <<enumeration>> {
+    tsp
+    tbp
+    cup
+    ml
```

Comment:
Not sure if enum is the best idea for units, cause there are a lot.

If you do want to go with enum, I think these should be added too

* Pint
* Gallon
* Quarts
* Litres
* Stick (for stuff like butter)

Reply:
How else could we collect all possible units? Noted on the pint, gallon, quarts, litres, sticks, etc.

Reply:
Maybe use `String`s for now, but keep a feature branch open that implements the `Unit` class properly